File: jeevan/H_Little_Girl_and_Maximum_Sum.py
n , q = map(int,input().split())
a = list(map(int,input().split()))
a.sort(reverse=True)
d = {}
for i in range(q):
    x,y = map(int,input().split())
    for i in range(x,y+1):
        if i in d:
            d[i]+=1
        else:
            d[i]=1

xx = list(d.values())
xx.sort(reverse=True)
ans = 0
for i in range(len(xx)):
    ans+=xx[i]*a[i]
print(ans)

File: jeevan/H_Little_Girl_and_Maximum_Sum.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <map>
using namespace std;
 
int main() {
    int n, q;
    cin >> n >> q;
    vector<int> a(n);
    for (int i = 0; i < n; ++i) cin >> a[i];
    sort(a.rbegin(), a.rend());
 
    vector<int> freq(n + 1, 0);
    for (int i = 0; i < q; ++i) {
        int x, y;
        cin >> x >> y;
        freq[x - 1]++;
        freq[y]--;
    }
 
    for (int i = 1; i < n; ++i) {
        freq[i] += freq[i - 1];
    }
 
    freq.pop_back();
    sort(freq.rbegin(), freq.rend());
 
    long long ans = 0;
    for (int i = 0; i < n; ++i) {
        ans += (long long)freq[i] * a[i];
    }
 
    cout << ans << endl;
    return 0;
}